distribution/services/update.go

```diff
@@ -0,0 +1,145 @@
+	httpClientsDetails := ur.DistDetails.CreateHttpClientDetails()
+	content, err := json.Marshal(releaseBundle)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	url := ur.DistDetails.GetUrl() + "api/v1/release_bundle/" + name + "/" + version
+	distrbutionServiceUtils.SetGpgPassphrase(gpgPassphrase, &httpClientsDetails.Headers)
+	artifactoryUtils.SetContentType("application/json", &httpClientsDetails.Headers)
+	resp, body, err := ur.client.SendPut(url, content, &httpClientsDetails)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return errorutils.CheckError(errors.New("Distribution response: " + resp.Status + "\n" + utils.IndentJson(body)))
+	}
+
+	log.Debug("Artifactory response: ", resp.Status)
+	log.Output(utils.IndentJson(body))
+	return errorutils.CheckError(err)
+}
+
+func CreateBundleBody(createBundleParams CreateUpdateReleaseBundleParams, dryRun bool) (*ReleaseBundleBody, error) {
+	var bundleQueries []BundleQuery
+	// Create release bundle queries
+	for _, specFile := range createBundleParams.SpecFiles {
+		if specFile.GetSpecType() != artifactoryUtils.AQL {
+			query, err := artifactoryUtils.CreateAqlBodyForSpecWithPattern(specFile)
+			if err != nil {
+				return nil, err
+			}
+			specFile.Aql = artifactoryUtils.Aql{ItemsFind: query}
+			aql := artifactoryUtils.BuildQueryFromSpecFile(specFile, artifactoryUtils.NONE)
+			bundleQueries = append(bundleQueries, BundleQuery{Aql: aql})
+		}
+	}
+
+	// Create release bundle struct
+	releaseBundleBody := &ReleaseBundleBody{
+		DryRun:            dryRun,
+		SignImmediately:   createBundleParams.SignImmediately,
+		StoringRepository: createBundleParams.StoringRepository,
+		Description:       createBundleParams.Description,
+		BundleSpec: BundleSpec{
+			Queries: bundleQueries,
+		},
+	}
+
+	// Add relese notes if needed
+	if createBundleParams.ReleaseNotes != "" {
+		releaseBundleBody.ReleaseNotes = &ReleaseNotes{
+			Syntax:  createBundleParams.ReleaseNotesSyntax,
+			Content: createBundleParams.ReleaseNotes,
+		}
+	}
+	return releaseBundleBody, nil
+}
+
+type ReleaseBundleBody struct {
+	DryRun            bool          `json:"dry_run"`
+	SignImmediately   bool          `json:"sign_immediately,omitempty"`
+	StoringRepository string        `json:"storing_repository,omitempty"`
+	Description       string        `json:"description,omitempty"`
+	ReleaseNotes      *ReleaseNotes `json:"release_notes,omitempty"`
+	BundleSpec        BundleSpec    `json:"spec"`
+}
+
+type ReleaseNotes struct {
+	Syntax  ReleaseNotesSyntax `json:"syntax,omitempty"`
+	Content string             `json:"content,omitempty"`
+}
+
+type BundleSpec struct {
+	Queries []BundleQuery `json:"queries"`
+}
+
+type BundleQuery struct {
+	QueryName string `json:"query_name,omitempty"`
+	Aql       string `json:"aql"`
+}
+
+type CreateUpdateReleaseBundleParams struct {
```

Comment:
I suggest splitting this struct into two structs - CreateReleaseBundleParams and CreateReleaseBundleParams.
Both can reuse the same internal struct, to avoid code duplication.